libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -164,7 +177,7 @@ class IRGenerationContext
 	/// The order and duplicates are irrelevant here (hence std::set rather than std::queue) as
 	/// long as the order of Yul functions in the generated code is deterministic and the same on
 	/// all platforms - which is a property guaranteed by MultiUseYulFunctionCollector.
-	std::set<FunctionDefinition const*> m_functionGenerationQueue;
+	DispatchSet m_functionGenerationQueue;
```

Comment:
Interesting! Did we generate the functions from there in order?

Reply:
Yes. `MultiUseYulFunctionCollector` takes care of that because it has its own map, with Yul function names as keys.

So we assumed it's fine because the functions themselves are placed in the bytecode in deterministic order but we did not consider the order of cases in the dispatch (which was determined by pointers in the set you see here).